log-appender/src/main/scala/io/phdata/pulse/Stream.scala

```diff
@@ -16,32 +16,51 @@
 
 package io.phdata.pulse
 
+import java.util.concurrent.Executors
+
 import io.phdata.pulse.log.{ HttpManager, JsonParser }
+import monix.execution.Scheduler
 import monix.reactive.subjects.ConcurrentSubject
 import monix.execution.Scheduler.Implicits.global
+import monix.reactive.OverflowStrategy
+import org.apache.log4j.helpers.LogLog
 import org.apache.log4j.spi.LoggingEvent
 
+import scala.concurrent.duration._
 import scala.concurrent.duration.FiniteDuration
+import scala.util.{ Failure, Success, Try }
+
+abstract class Stream[E](flushDuration: FiniteDuration, flushSize: Int, maxBuffer: Int) {
 
-abstract class Stream[E](flushDuration: FiniteDuration, flushSize: Int) {
-  val subject = ConcurrentSubject.publish[E]
+  val overflowStragegy = OverflowStrategy.DropNewAndSignal(maxBuffer, (_: Long) => None)
+  val subject          = ConcurrentSubject.publish[E](overflowStragegy)
 
   subject
-    .bufferTimedAndCounted(flushDuration, flushSize)
-    .foreach(save)
+    .bufferTimedAndCounted(2 seconds, flushSize)
```

Comment:
should `flushDuration` be used instead of the `2 seconds`?

Reply:
Indeed it should